src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -187,5 +186,27 @@ public void SetAccessControl_FileStream_FileSecurity_Success()
                 FileSystemAclExtensions.SetAccessControl(fileStream, fileSecurity);
             }
         }
+
+        [Fact]
+        public void Create_FileStream_From_FileInfo()
+        {
+            using var directory = new TempDirectory();
+            string path = Path.Combine(directory.Path, "file.txt");
+            FileInfo info = new FileInfo(path);
+            FileSecurity security = new FileSecurity();
+            using FileStream stream = info.Create(FileMode.CreateNew, FileSystemRights.FullControl, FileShare.ReadWrite, 1, FileOptions.None, security);
+            Assert.True(File.Exists(path));
```

Comment:
Please read back the ACL's . It would be not good to set the wrong ones or fail to set them :)

Reply:
Good point. That's the whole reason of this PR. 😄

Reply:
A good choice would be to give full rights to "Users" as you'll be able to validate it reasonably easily.

Reply:
Note that, if we get this wrong (eg set the wrong ACL) then servicing our code won't help. The ACL's will already have been written to filesystems everywhere. It will be quite bad.

Reply:
@danmosemsft I added verifications in the unit tests in case you want to take another look :)